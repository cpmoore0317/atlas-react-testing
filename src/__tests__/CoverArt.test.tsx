import { render } from "@testing-library/react";
import CoverArt from "@/components/CoverArt";
import { expect, it, describe } from "vitest";

describe('CoverArt Component', () => {
    it('renders with image URL', () => {
        const { container } = render(<CoverArt src="https://example.com/cover.jpg" alt="Cover Art" />);
        expect(container).toMatchSnapshot();
    });

    it('renders without image URL (fallback)', () => {
        const { container } = render(<CoverArt src="" alt="Cover Art" />);
        expect(container).toMatchSnapshot();
    });
});
