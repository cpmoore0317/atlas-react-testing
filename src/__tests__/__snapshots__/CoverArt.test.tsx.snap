// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt Component > renders with image URL 1`] = `
<div>
  <img
    alt="Cover Art"
    class="w-full h-auto rounded-md object-cover"
    src="https://example.com/cover.jpg"
  />
</div>
`;

exports[`CoverArt Component > renders without image URL (fallback) 1`] = `
<div>
  <img
    alt="Cover Art"
    class="w-full h-auto rounded-md object-cover"
    src=""
  />
</div>
`;
